### PR TITLE
Task/configure mandatory fields

### PR DESCRIPTION
**Pull request by jbertel2**
Since we have found many ways in which mandatory fields in ContentStack can end up being empty during a gatsby build, we prefer to have an option to not break the build when null is returned from a resolver for a mandatory field in CS.

Ways to have empty mandatory fields include:

1. Having mandatory references which are not published because only direct references are published automatically, but not indirect ones
2. Having not published all locales
3. Adding mandatory fields to content types with existing entries

### CODE REVIEW

**Comment by ninadhatkar** [comment]
@jbertel2 thanks for your contribution we will review the pull request and keep you posted.

**Comment by jbertel2** [comment]
@ninadhatkar Willing to review this PR, please?

**Comment by ninadhatkar** [comment]
@jbertel2 We are working on scheduling this. Will keep you posted.

File: src/normalize.js [hunk 1]
```diff
@@ -449,6 +451,18 @@ exports.extendSchemaWithDefaultEntryFields = schema => {
     multiple: false,
     mandatory: false,
   });
+  schema.push({
+    data_type: 'isodate',
```
Comment (ninadhatkar) [inline src/normalize.js:455]:
```suggestion
    data_type: 'Date',
```

**Comment by vkalta** [comment]
> Since we have found many ways in which mandatory fields in ContentStack can end up being empty during a gatsby build, we prefer to have an option to not break the build when null is returned from a resolver for a mandatory field in CS.
> 
> Ways to have empty mandatory fields include:
> 
> 1. Having mandatory references which are not published because only direct references are published automatically, but not indirect ones
> 2. Having not published all locales
> 3. Adding mandatory fields to content types with existing entries

Hi @jbertel2 , I managed to the test the PR for 3rd case above, although I was not able reproduce 1st and 2nd cases. Is it possible for you to elaborate on these cases?

Thanks.

**Comment by vkalta** [comment]
@jbertel2 The changes in this PR are released and you can find them on version 2.3.7.  Thanks for your contribution.

**Pull request merged**